src/plugins/pendo/agent/filters.ts

```diff
@@ -0,0 +1,225 @@
+import { logger } from 'lib/logger'
+
+import { PendoGuideFilterError, PendoGuideRequestError, PendoGuideResponseError } from './errors'
+import type { PendoEnv } from './types'
+
+export const expectedResponseKeys = [
+  'autoOrdering',
+  'designerEnabled',
+  'features',
+  'globalJsUrl',
+  'guideCssUrl',
+  'guideWidget',
+  'guides',
+  'lastGuideStepSeen',
+  'normalizedUrl',
+  'preventCodeInjection',
+  'segmentFlags',
+  'throttling',
+  'props',
+  'type',
+  'children',
+  'latestDismissedAutoAt',
+]
+
+const moduleLogger = logger.child({ namespace: ['Plugin', 'Pendo', 'Agent', 'Filters'] })
+
+export function filterGuideTag(
+  this: PendoEnv,
```

Comment:
Ditto, feel free to resolve all of these if there's some automagic I didn't get here